mitiq/utils.py

```diff
@@ -133,3 +133,50 @@ def _are_close_dict(dict_a: Dict[Any, Any], dict_b: Dict[Any, Any]) -> bool:
         if not np.isclose(dict_b[ka], va):
             return False
     return True
+
+
+def _get_base_gate(gate: EigenGate) -> EigenGate:
+    BASE_GATES = [ZPowGate, HPowGate, XPowGate, YPowGate, CXPowGate, CZPowGate]
+
+    for base_gate in BASE_GATES:
+        if isinstance(gate, base_gate):
+            return cast(EigenGate, base_gate)
+    raise GateTypeException(
+        "Must have circuit be made of rotation gates. "
+        "Your gate {} may not be supported".format(gate)
+    )
```

Comment:
I'm not entirely clear on this method. One function is to check that the gate type is supported, but why does the `cast` need to happen? If the `gate` is already an instance (or at least an instance of a subtype) of the base_gate why is the cast needed?

Apologies if this is obvious, but it may at least be worth an explanatory comment on the cast for future reference